### PR TITLE
driver score view

### PR DESCRIPTION
**Pull request by RingerJK**
Don't merge it. Waiting for other changes

### CODE REVIEW

File: app/src/main/java/com/mapbox/vision/examples/CircleView.kt [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+class CircleView
+@JvmOverloads
+constructor(
+    context: Context,
+    attrs: AttributeSet? = null,
+    defStyleAttr: Int = 0,
+    defStyleRes: Int = 0
+) : View(context, attrs, defStyleAttr, defStyleRes) {
+
+    private val stokePx = context.dpToPx(2f)
```
Comment (yunikkk) [inline app/src/main/java/com/mapbox/vision/examples/CircleView.kt]: strokePx?

Comment (RingerJK) [reply app/src/main/java/com/mapbox/vision/examples/CircleView.kt]: yep

File: app/src/main/res/layout/activity_main.xml [hunk 2]
```diff
@@ -323,20 +336,20 @@
...
             android:orientation="horizontal"
-            app:layout_constraintGuide_percent="0.56" />
+            app:layout_constraintGuide_percent="0.56"/>
 
         <View
             android:layout_width="match_parent"
             android:layout_height="1dp"
             android:background="@drawable/menu_center_line_gradient_bg"
-            app:layout_constraintTop_toBottomOf="@+id/guideline" />
+            app:layout_constraintTop_toBottomOf="@+id/guideline"/>
```
Comment (yunikkk) [inline app/src/main/res/layout/activity_main.xml]: Lets rollback this spaces?

File: app/src/main/res/layout/activity_main.xml [hunk 3]
```diff
@@ -431,10 +444,42 @@
...
+                    android:textStyle="bold"/>
 
             </LinearLayout>
 
+            <!--<View-->
+                <!--android:layout_width="1dp"-->
+                <!--android:layout_height="match_parent"-->
+                <!--android:background="@drawable/menu_line_gradient_rotated_bg"/>-->
+
+            <!--<LinearLayout-->
```
Comment (yunikkk) [inline app/src/main/res/layout/activity_main.xml]: Should we hide it instead of commenting?

File: app/src/main/res/layout/activity_main.xml (hunk 2 above)
Comment (RingerJK) [reply app/src/main/res/layout/activity_main.xml]: I think it's xml codeStyle for our project

File: app/src/main/res/layout/activity_main.xml (hunk 3 above)
Comment (RingerJK) [reply app/src/main/res/layout/activity_main.xml]: agree, done

File: app/src/main/java/com/mapbox/vision/examples/ScoreProgressView.kt [hunk 4]
```diff
@@ -0,0 +1,203 @@
...
+                    showWarningScore(nextScore.score)
+                }
+            }
+
+        }, delayMills)
+    }
+
+    private fun redrawProgress() {
+        progress_gray.layoutParams.width =
+            (progressGradientWidth * (1.01f - currentScore.toFloat() / SCORE_MAX)).toInt()
```
Comment (yunikkk) [inline app/src/main/java/com/mapbox/vision/examples/ScoreProgressView.kt]: 1.01 ?

File: app/src/main/java/com/mapbox/vision/examples/CircleView.kt (hunk 1 above)
Comment (yunikkk) [reply app/src/main/java/com/mapbox/vision/examples/CircleView.kt]: I mean here is a typo, stroke and not stoke ^_^

Comment (RingerJK) [reply app/src/main/java/com/mapbox/vision/examples/CircleView.kt]: fix typo :)

File: app/src/main/java/com/mapbox/vision/examples/ScoreProgressView.kt (hunk 4 above)
Comment (RingerJK) [reply app/src/main/java/com/mapbox/vision/examples/ScoreProgressView.kt]: yes, it's just to not check corner case when `progressGradientWidth*0`

Comment (yunikkk) [reply app/src/main/java/com/mapbox/vision/examples/ScoreProgressView.kt]: Do you think its worth to leave comment about it?

Comment (RingerJK) [reply app/src/main/java/com/mapbox/vision/examples/ScoreProgressView.kt]: refactored

**Comment by yunikkk** [comment]
@RingerJK could you squash commits pls?

**Comment by RingerJK** [comment]
@yunikkk sure, np

**Pull request merged**